tests/e2e/simpleevmtest.go

```diff
@@ -0,0 +1,463 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc"
+
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+	"github.com/oasisprotocol/oasis-core/go/common/logging"
+
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/client"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/testing"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+
+	"github.com/oasisprotocol/oasis-sdk/tests/e2e/txgen"
+)
+
+// The evmCreateTx type must match the CreateTx type from the evm module types
+// in runtime-sdk/src/modules/evm/types.rs.
+type evmCreateTx struct {
+	Value    []byte `json:"value"`
+	InitCode []byte `json:"init_code"`
+	GasPrice []byte `json:"gas_price"`
+	GasLimit uint64 `json:"gas_limit"`
+}
+
+// The evmCallTx type must match the CallTx type from the evm module types
+// in runtime-sdk/src/modules/evm/types.rs.
+type evmCallTx struct {
+	Address  []byte `json:"address"`
+	Value    []byte `json:"value"`
+	Data     []byte `json:"data"`
+	GasPrice []byte `json:"gas_price"`
+	GasLimit uint64 `json:"gas_limit"`
+}
+
+// The evmPeekStorageQuery type must match the PeekStorageQuery type from the
+// evm module types in runtime-sdk/src/modules/evm/types.rs.
+type evmPeekStorageQuery struct {
+	Address []byte `json:"address"`
+	Index   []byte `json:"index"`
+}
+
+// The evmPeekCodeQuery type must match the PeekCodeQuery type from the
+// evm module types in runtime-sdk/src/modules/evm/types.rs.
+type evmPeekCodeQuery struct {
+	Address []byte `json:"address"`
+}
+
+func evmCreate(ctx context.Context, rtc client.RuntimeClient, signer signature.Signer, tx evmCreateTx) ([]byte, error) {
```

Comment:
Also these functions would be useful to have in the client-sdk, see how it's done in the contracts module PR.